src/app/app.component.ts

```diff
@@ -80,9 +84,52 @@ export class AppComponent implements OnInit {
       });
   }
 
+  toggle(value: string) {
+    if (value === 'toggle' && !this.window.isXS) {
+      this.sideMenu = !this.sideMenu;
+      this.unPin = this.sideMenu ? 'Hide Menu' : 'Show Menu';
+      this.resizeMenu('toggle');
+      return;
+    }
+    if (!this.sideMenu) {
+      this.isPinned = value === 'enter';
+    }
+  }
+
+  resizeMenu(type: string) {
```

Comment:
No way to do this without hack?

Reply:
yeah, I have tried to do with css but it didn't worked, that the reason used kind of javascript related to all the scenario

Reply:
It used to work in master without all of this, did you check that out?

Reply:
yes I did check that and found that the sidemenu used was bootstrap and now it is material  one, and I don't see the same properties available in material sidemenu. Also when I toggling the sidmenu to decrease the width the weird thing is it taking that space, so resolve those issues I used the above stuff.